### ISSUE TITLE
Thread jobs (`Start-ThreadJob`), unlike child process-based jobs (`Start-Job`), do not support `Wait-Debugger`, causes hang when calling `Debug-Job`

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#21257]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Note: 
* With the `ThreadJob` / `Microsoft.PowerShell.ThreadJob` module confusion, it's unclear to me where to report relevant issues, so I'm doing it here for now.

---

Note:

* The following works fine if you use `Start-Job` instead of `Start-ThreadJob`
* The thread job never enters the `AtBreakPoint` state and after startup reports itself as `Running`, indefinitely.
* If you wait for `Running` so you can call `Debug-Job`, the latter hangs too.

```powershell
# Start a job that calls Wait-Debugger right away.
$jb = Start-ThreadJob {
  Wait-Debugger
  "after Wait-Debugger"
}

try {

  Write-Verbose -Verbose 'Waiting for job to be in the ''AtBreakPoint'' state...'
  while ($jb.State -notin  'AtBreakPoint') { Start-Sleep -Milliseconds 100 }

  Write-Verbose -Verbose 'Starting debugging...'
  $jb | Debug-Job
  $jb | Receive-Job -Wait -AutoRemoveJob
  $jb = $null
}
finally {
  if ($jb) { $jb | Remove-Job -Force }
}
```

### Expected behavior

```console
VERBOSE: Waiting for job to be in the 'AtBreakPoint' state...
VERBOSE: Starting debugging...
At line:2 char:3
+   Wait-Debugger
+   ~~~~~~~~~~~~~
[DBG]: [Job9]: PS /Users/mklement/Desktop/pg>>

That is, the `Debug-Job` call should trigger breaking into the debugger at the `Wait-Debugger` statement in the thread job.
```

### Actual behavior

```console
VERBOSE: Waiting for job to be in the 'AtBreakPoint' state...
```

### Environment data

```powershell
PowerShell 7.5.0-preview.1, ThreadJob 2.0.3
```

**Comment by kilasuit** [comment]
ThreadJob module is in [the Thread job Repo](https://github.com/PowerShell/ThreadJob/)

however assigning to @JamesWTruher as to move this issue across to that repo (to save rasing another one in that repo & linking to this one)

**Comment by kilasuit** [comment]
Yeah that was the original repo, when Paul was part of the team but he's now retired and that's why the PowerShell one is the right one going forward.
